Can't use non latin1 characters in basic auth

Hi!
I've tried to use webtest-plus with utf-8 encoded password and it fails, trying to encode that data to `latin1` first. 
As far as I understand basic auth, the point of encoding to base64 is to be able to pass any binary data as the value, so I'm not sure if `encode` / `decode` to latin1 step is necessary.  

```
username = 'John', password = '\xc5\xbc'

    def _basic_auth_str(username, password):
        """Returns a Basic Auth string."""
>       return 'Basic ' + b64encode(('%s:%s' % (username, password)).encode('latin1')).strip().decode('latin1')
E       UnicodeDecodeError: 'ascii' codec can't decode byte 0xc5 in position 5: ordinal not in range(128)

/usr/local/lib/python2.7/site-packages/webtest_plus/app.py:13: UnicodeDecodeError

```

--- a/webtest_plus/app.py
+++ b/webtest_plus/app.py
@@ -5,12 +5,20 @@ import webtest
 from webtest import utils
 
 from webtest_plus.response import TestResponse
-from webtest_plus.compat import PY2, binary_type
+from webtest_plus.compat import PY2, binary_type, unicode
 
 
 def _basic_auth_str(username, password):
     """Returns a Basic Auth string."""
-    return 'Basic ' + b64encode(('%s:%s' % (username, password)).encode('latin1')).strip().decode('latin1')
+    if isinstance(username, unicode):
+        username = username.encode('latin1')
+
+    if isinstance(password, unicode):
+        password = password.encode('latin1')
+
+    return b'Basic ' + (
+        b64encode(b':'.join((username, password))).strip()
+    )
 
 
 def _add_auth(auth, headers, auth_type='basic'):

